app/Http/Controllers/Service/Admin/VouchersController.php

```diff
@@ -45,69 +41,109 @@ public function create()
-        Validator::make($request->all(), $voucher_rules, $messages)->validate();
-
-        $codes = range($request['start'], $request['end']);
-
-        $sponsor = Sponsor::find($request['sponsor_id'])->first();
-        $shortcode = $sponsor->shortcode;
+        // Setup some variables
+        $input = $request->all();
+        $user_id = auth()->id();
+        $user_type = class_basename(auth()->user());
+        $sponsor = Sponsor::findOrFail($input['sponsor_id']);
         $now_time = Carbon::now();
-        foreach ($codes as $c) {
-            $v = new Voucher();
-            $v->code = $shortcode . $c;
-            $v->sponsor_id = $request['sponsor_id'];
-            $v->currentstate = 'requested';
-            $v->created_at = $now_time;
-            $v->updated_at = $now_time;
-            $new_vouchers[] = $v->attributesToArray();
+        $maxStep = 1000;
+
+        // Calulate the number of codes we need
+        $numCodes = $input['end'] - $input['start'];
+
+        $step = null;
+        if ($numCodes > 1) {
+            // Calculate the step, max = $maxStep.
+            $step = ($numCodes < $maxStep)
+                ? $numCodes
+                : $maxStep;
+        }
+
+        // Setup the chunks
+        $chunks = range(
+            $input['start'],
+            $input['end'],
+            $step
+        );
+
+        // Add the range to the end.
+        if (!in_array($input['end'], $chunks)) {
+            $chunks[] = $input['end'];
         }
-        // batch insert.
-        // Todo : there's NO "this voucher already exists" checking!!
-        Voucher::insert($new_vouchers);
-
-        $vouchers = Voucher::where('created_at', '=', $now_time)
-            ->where('updated_at', '=', $now_time)
-            ->where('currentstate', '=', 'requested')
-            ->get()
-        ;
-
-        // batch progress
-        foreach ($vouchers as $voucher) {
-            $voucher->applyTransition('order');
-            $voucher->applyTransition('print');
+
+        // For each chunk, create the integers in that set.
+        foreach ($chunks as $k => $chunkStart) {
+            // Reset New Vouchers
+            $new_vouchers = [];
+
+            $chunkEnd = (isset($chunks[$k + 1]))
+                ? $chunks[$k + 1] - 1
+                : $input['end'];
+            $currentChunk = range($chunkStart, $chunkEnd);
+
+            foreach ($currentChunk as $c) {
+                $v = new Voucher();
+                $v->code = $sponsor->shortcode . $c;
+                $v->sponsor_id = $sponsor->id;
+                // Set straight to printed; we're faking the process for speed.
+                $v->currentstate = 'printed';
+                $v->created_at = $now_time;
+                $v->updated_at = $now_time;
+                $new_vouchers[] = $v->attributesToArray();
+                unset($v);
+            }
+            // Batch insert raw vouchers.
+            // Todo : there's NO "this voucher already exists" checking!!
```

Comment:
How likely is it that this will be an issue?

Reply:
Ugh, could be a serious problem, BUT i think we'll fix it in the next 24 hours OR in the next dev sprint. ARC have promised to be careful.

Reply:
:grimacing: It'll be ok.